### PR TITLE
fix(ProfilesCache): Clear object caches before refresh

### LINKED ISSUE #2395: zowe-explorer-api ProfilesCache holding onto stale service profile from extenders when it is deleted from team config
We are currently contributing our own zowe service profiles (type zOpenDebug) using zowe-explorer-api. Our code has it's own watchers set up on the team config files. However, when all profiles of type zOpenDebug are removed from the team config file, the ZE-api ProfilesCache will still contain the removed profile. Note that aside from this case, the ProfilesCache seems to work as expected when multiple zOpenDebug profiles are added/deleted. It's only when they are ALL removed, that this bug occurs.

1. Create a new zowe profile type using zowe-explorer-api. 
2. Create a zowe team config file if one isn't already present.
3. Delete All instances of the new profile type from the team config file. 
4. call ProfileCache.refresh(). Despite profiles being removed from team config file, ProfilesCache will still return a reference to a profile.

**Expected behavior**

The ProfileCache should return an empty list for the new zowe profile type if they are all removed from the team config

**Desktop (please complete the following information):**

- OS: Windows 11
- Zowe Explorer Version: we are using @zowe/zowe-explorer-api@2.8.0

### PR DESCRIPTION
**Pull request by traeok**
## Proposed changes

Resolves #2395 

## Release Notes

Milestone: 2.15.2

Changelog:

- Fixed an issue where the `ProfilesCache` class would retain old service profiles, even if they were removed from the team config.

## Types of changes

What types of changes does your code introduce to Zowe Explorer?
_Put an `x` in the boxes that apply_

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Updates to Documentation or Tests (if none of the other choices apply)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This checklist will be used as reference for both the contributor and the reviewer_

- [x] I have read the [CONTRIBUTOR GUIDANCE](https://github.com/zowe/zowe-explorer-vscode/wiki/Best-Practices:-Contributor-Guidance) wiki
- [x] PR title follows [Conventional Commits Guidelines](https://www.conventionalcommits.org/en/v1.0.0-beta.2/)
- [x] PR Description is included
- [ ] gif or screenshot is included if visual changes are made
- [x] `yarn workspace vscode-extension-for-zowe vscode:prepublish` has been executed
- [x] All checks have passed (DCO, Jenkins and Code Coverage)
- [x] I have added unit test and it is passing
- [ ] I have added integration test and it is passing
- [ ] There is coverage for the code that I have added
- [x] I have tested it manually and there are no regressions found

### CODE REVIEW

**Review by zFernand0 — DISMISSED** [review]
This issue isn't a high priority one, but given the size of the fix, I don't mind if this PR is combined with https://github.com/zowe/zowe-explorer-vscode/pull/2760 

Changes LGTM! 😋 

Please don't forget to open a PR from `maintenance` to `main` once the 2.15.2 is out.

I don't believe I've seen one for the 2.15.1 patch yet 😋

File: packages/zowe-explorer-api/src/profiles/ProfilesCache.ts [hunk 1]
```diff
@@ -194,6 +194,7 @@ export class ProfilesCache {
     public async refresh(apiRegister?: ZoweExplorerApi.IApiRegisterClient): Promise<void> {
         this.allProfiles = [];
         this.allTypes = [];
+        this.profilesByType.clear();
```
Comment (t1m0thyj) [inline packages/zowe-explorer-api/src/profiles/ProfilesCache.ts:197]: Thanks for adding this 🙂 Should we also clear the `defaultProfileByType` map?

Comment (traeok) [reply packages/zowe-explorer-api/src/profiles/ProfilesCache.ts:197]: No problem - and yes, thanks for catching that, addressed in `ce52b02`

**Review by t1m0thyj — APPROVED** [review]
LGTM, thanks @traeok!

**Review by zFernand0 — APPROVED** [review]
LGTM! 😋

**Pull request merged**